modules/deployment-model-impl/src/main/java/org/mule/runtime/module/deployment/impl/internal/policy/ArtifactExtensionManagerFactory.java

```diff
@@ -77,19 +79,15 @@ public ExtensionManager create(MuleContext muleContext) {
         Map<String, Object> params = new HashMap<>();
         params.put(TYPE_PROPERTY_NAME, extensionManifest.getDescriberManifest().getProperties().get("type"));
         params.put(VERSION, extensionManifest.getVersion());
-
-        extensions.add(new JavaExtensionModelLoader()
-            .loadExtensionModel(artifactPlugin.getArtifactClassLoader().getClassLoader(),
-                                getDefault(extensions),
-                                params));
+        ClassLoader cl = artifactPlugin.getArtifactClassLoader().getClassLoader();
+        extensions.add(new DefaultJavaExtensionModelLoader().loadExtensionModel(cl, getDefault(emptySet()), params));
```

Comment:
that doesn't make it right. Passing an emptySet in there might be a problem in some cases. For smart connectors is fine, but it would just pass the extensions list as it used to